description: fix signedness error when built with gcc that affects safe browsing
origin: https://bugs.gentoo.org/show_bug.cgi?id=674504
author: Maciej S. Szmigiero
bug: http://bugs.debian.org/914886

--- a/base/strings/char_traits.h
+++ b/base/strings/char_traits.h
@@ -67,9 +67,9 @@ constexpr int CharTraits<char>::compare(
   return __builtin_memcmp(s1, s2, n);
 #else
   for (; n; --n, ++s1, ++s2) {
-    if (*s1 < *s2)
+    if ((unsigned char)*s1 < (unsigned char)*s2)
       return -1;
-    if (*s1 > *s2)
+    if ((unsigned char)*s1 > (unsigned char)*s2)
       return 1;
   }
   return 0;
